contracts/instruments/StakedPut.sol

```diff
@@ -0,0 +1,318 @@
+
+    struct BuyInstrumentParams {
+        address paymentToken;
+        uint256 putStrikePrice;
+        uint256 optionAmount;
+        uint256 putMaxCost;
+        uint256 expiry;
+        uint256 lpAmt;
+        string exchangeName;
+        uint256 tradeAmt;
+        uint256 minWbtcAmtOut;
+        uint256 minDiggAmtOut;
+    }
+
+    using AmmAdapter for IAmmAdapter;
+    using ProtocolAdapter for IProtocolAdapter;
+    using SafeERC20 for IERC20;
+    using SafeMath for uint256;
+
+    IRibbonFactory public immutable factory;
+    IProtocolAdapter public immutable adapter;
+    IAmmAdapter public immutable iUniswapAdapter;
+    IUniswapV2Pair public immutable ethWbtcPair;
+    IHegicOptions public immutable options;
+
+    address payable public uniswapAdapterAddress;
+    string private constant adapterName = "HEGIC";
+    string private constant instrumentName = "wbtc/digg-staked-put";
+    address private constant _WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+    uint256 public constant slippageBuffer = 1000;
+    uint256 private constant month = 172800;
+    bytes32 private constant uniswapHash = keccak256(abi.encodePacked("UNISWAP"));
+    string public constant venue = 'HEGIC';
+    uint8 public constant venueID = 1;
+    ProtocolAdapterTypes.OptionType public constant optionType = ProtocolAdapterTypes.OptionType.Put;
+    address public immutable ethAddress;
+    address public immutable wbtcAddress;
+    address public immutable underlying;
+    address public immutable strikeAsset;
+    address public immutable collateralAsset;
+    address public immutable optionsAddress;
+
+
+    constructor(address _factory, address payable _uniswapAdapterAddress, address ethWbtcPairAddr, address _ethAddress, address _wbtcAddress, address _wbtcOptionsAddress, address _collateralAsset) {
+        require(_factory != address(0), "!_factory");
+        IRibbonFactory factoryInstance = IRibbonFactory(_factory);
+
+        ethAddress = _ethAddress;
+        wbtcAddress = _wbtcAddress;
+        underlying = _wbtcAddress;
+        strikeAsset = _wbtcAddress;
+        collateralAsset = _collateralAsset;
+
+        iUniswapAdapter = IAmmAdapter(_uniswapAdapterAddress);
+        uniswapAdapterAddress = _uniswapAdapterAddress;
+        ethWbtcPair = IUniswapV2Pair(ethWbtcPairAddr);
+        address adapterAddress = factoryInstance.getAdapter(adapterName);
+        require(adapterAddress != address(0), "Adapter not set");
+        options = IHegicOptions(_wbtcOptionsAddress);
+
+        factory = factoryInstance;
+        adapter = IProtocolAdapter(adapterAddress);
+        optionsAddress = _wbtcOptionsAddress;
+
+    }
+
+    function initialize() external initializer {
+    }
+
+    function getName() public pure returns(string memory)
+    {
+        return instrumentName;
+    }
+
+    function getCurrentPrice() public view returns(uint256)
+    {
+        AggregatorV3Interface priceProvider = AggregatorV3Interface(options.priceProvider());

```

Comment:
Since we are only using the `IHegicOptions` to get the `priceProvider` variable, we can just pass in the price feed address into the contract.

This helps us avoid being tightly coupled with hegic